File: SkaiLab.Invoice.Models/Menu.cs
using System.Collections.Generic;

namespace SkaiLab.Invoice.Models
{
    public class Menu
    {
        public string Name { get; set; }
        public string IconClass { get; set; }
        public bool IsActive { get; set; }
        public List<Menu> MenuItems { get; set; }
        public string RouteLink { get; set; }
    }
}

File: SkaiLab.Invoice/ClientApp/src/app/models/menu.ts
export class Menu{
    name:string;
    iconClass:string;
    isActive:boolean;
    menuItems:Menu[];
    routeLink:string;
    constructor(){
        this.name="";
        this.iconClass="";
        this.isActive=false;
        this.menuItems=[];
        this.routeLink="";
    }
}